general/cuda.hpp

```diff
@@ -19,6 +19,7 @@
 #define MFEM_CUDA_BLOCKS 256
 
 #ifdef MFEM_USE_CUDA
+#include <cuda.h>
```

Comment:
We moved the system headers to `backends.hpp`, so that it'll be easier to switch to another backend (`hip`).
Would that work for you to include it?

Reply:
I believe `cuda.h` belongs here since `general/cuda.hpp` uses CUDA_VERSION

Reply:
I agree that there is one `include` that has sneaked in the `linalg/sparsmat.cpp` - but the goal of `backends.hpp` is to add all required system headers for the device `MFEM` has been configured for (`MFEM_USE_CUDA`, `MFEM_USE_HIP`, ...). 

When we'll want to add the `PETSc` `HIP` support, nothing will need to change - hopefully.

The `__CUDA_ARCH__` macro will be set by the `nvcc` compiler.